### PR TITLE
[Delivers #79483250] document the architecture

### PR DESCRIPTION
**Pull request by afeld**
...among other things. Also made the app ignore the SMTP settings in development mode, since they aren't used.

### CODE REVIEW

**Comment by afeld** [comment]
Ready for review!

**Comment by dlapiduz** [comment]
:+1: :+1: :+1: :+1: :+1: :+1: :+1:

File: doc/overview.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Use cases
+
+This application contains code for several independent but similar use cases. Users will generally be segmented into one use case or another in terms of how the Proposals are initiated, though the approval workflow is (largely) the same.
+
+### [National Capitol Region (NCR) service centers](http://www.gsa.gov/portal/category/21528)
+
+The NCR use case was built around GSA service centers (paint shops, landscapers, etc.) needing approvals for their superiors and various budget officials for credit card purchases. They use the "linear" workflow described [above](#proposal-flows):
+
+1. The requester logs in via MyUSA.
```
Comment (phirefly) [inline doc/overview.md]: Do we need to mention that users currently need to be whitelisted?

Comment (afeld) [inline doc/overview.md]: Not sure what you mean... I don't know of a user whitelist on either the MyUSA or C2 side.

Comment (afeld) [inline doc/overview.md]: Ah ok, you're right. Where is the whitelist specified?

**Comment by cmc333333** [comment]
:+1:

**Comment by afeld** [comment]
Added a note about the whitelisting. Cool if we merge sooner than later, with the assumption that the docs will always need improvement?

**Pull request merged**